plugin/src/main/java/org/wildfly/plugin/server/AbstractServerStartMojo.java

```diff
@@ -113,190 +78,27 @@ public abstract class AbstractServerStartMojo extends AbstractServerConnection {
-    @Parameter(property = "wildfly.debug", defaultValue = "false")
-    private boolean debug;
-
-    /**
-     * Sets the hostname to listen on for debugging. An {@code *} means all hosts.
-     */
-    @Parameter(property = "wildfly.debug.host", defaultValue = "*")
-    private String debugHost;
-
-    /**
-     * Sets the port the debugger should listen on.
-     */
-    @Parameter(property = "wildfly.debug.port", defaultValue = "8787")
-    private int debugPort;
-
-    /**
-     * Indicates whether the server should suspend itself until a debugger is attached.
-     */
-    @Parameter(property = "wildfly.debug.suspend", defaultValue = "false")
-    private boolean debugSuspend;
-
-    /**
-     * The path to the system properties file to load.
-     */
-    @Parameter(alias = "properties-file", property = PropertyNames.PROPERTIES_FILE)
-    private String propertiesFile;
-
-    /**
-     * The timeout value to use when starting the server.
-     */
-    @Parameter(alias = "startup-timeout", defaultValue = "60", property = PropertyNames.STARTUP_TIMEOUT)
-    private long startupTimeout;
-
-    /**
-     * The arguments to be passed to the server.
-     */
-    @Parameter(alias = "server-args", property = PropertyNames.SERVER_ARGS)
-    private String[] serverArgs;
-
-    /**
-     * Set to {@code true} if you want to skip this goal, otherwise {@code false}.
-     */
-    @Parameter(defaultValue = "false", property = PropertyNames.SKIP)
-    protected boolean skip;
+    protected String[] moduleOptions;
 
     /**
      * The users to add to the server.
      */
     @Parameter(alias = "add-user", property = "wildfly.add-user")
     private AddUser addUser;
 
-    /**
-     * Specifies the environment variables to be passed to the process being started.
-     * <div>
-     *
-     * <pre>
-     * &lt;env&gt;
-     *     &lt;HOME&gt;/home/wildfly/&lt;/HOME&gt;
-     * &lt;/env&gt;
-     * </pre>
-     *
-     * </div>
-     */
-    @Parameter
-    private Map<String, String> env;
-
-    private final AtomicBoolean initialized = new AtomicBoolean();
-
-    protected MavenRepoManager mavenRepoManager;
-
-    protected void init() throws MojoExecutionException {
-        // Setting the mavenRepoManager is not thread-safe, however creating it more than once won't hurt anything
-        if (initialized.compareAndSet(false, true)) {
-            MavenRepositoriesEnricher.enrich(mavenSession, project, repositories);
-            mavenRepoManager = createMavenRepoManager();
-        }
-    }
-
-    protected MavenRepoManager createMavenRepoManager() throws MojoExecutionException {
-        return new MavenArtifactRepositoryManager(repoSystem, session, repositories);
-    }
-
-    protected ServerContext startServer(final ServerType serverType) throws MojoExecutionException, MojoFailureException {
-        final Log log = getLog();
-        init();
-
+    @Override
+    protected Path getServer() throws MojoExecutionException, MojoFailureException {
```

Comment:
I think we should call this `getServerHome()` or `getJBossHome()`.

Reply:
Done thanks.